src/controller/java/AndroidDeviceControllerWrapper.h

```diff
@@ -24,6 +24,8 @@
 
 #include <jni.h>
 
+#include <app/icd/client/CheckInHandler.h>
+#include <app/icd/client/DefaultCheckInDelegate.h>
```

Comment:
You should not use DefaultCheckInDelegate in android chip-tool, you should create a new CheckInDelegate, 
expose OnCheckInComplete to java/kotlin application.
Expose OnKeyRefreshNeeded to java/kotlin application so that symmetric token generation can happen in same place for both commissioning and keyRefreshStage, we also need make sure it is unique
Expose OnKeyRefreshDone to the java/kotlin application.

Reply:
based upon offline chat, the android CheckInDelegate would be follow-up